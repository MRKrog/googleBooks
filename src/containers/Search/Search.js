import React, {Component} from 'react';
import PropTypes from 'prop-types';

export class Search extends Component {
  constructor(props){
    super(props);
    this.state = {
      search: ''
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { search } = this.state;
    if(search.length > 0){
      this.props.handleSearch(search)
      this.setState({search: ''})
    }
  }

  handleInput = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    })
  }

  render() {
    return (
      <div className="Search">
        <section className="searchTitle">
           <h2>Google Books</h2>
        </section>
        <form className="searchForm">
           <input
             type="search"
             placeholder="Enter Your Search"
             name="search"
             value={this.state.search}
             onChange={this.handleInput}
           />
           <button type="submit" className="btnClick" onClick={this.handleSubmit}>Search</button>
         </form>
      </div>
    );
  }
}

Search.propTypes = {
  handleSearch: PropTypes.func,
}
