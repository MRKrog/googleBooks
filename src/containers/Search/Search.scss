.Search {
  display: flex;
  flex-direction: column;
  width: 40%;
  align-items: flex-start;
  padding: 1em 1em 0 1em;

  .searchTitle {
    font-family: $fontOswald;
    font-size: 2em;
    color: #fff;

    h2 {
      font-weight: 100;
      margin: 0;
      text-transform: uppercase;
      font-size: 1.2em;
      padding-bottom: 10px;
    }
  }

  .searchForm {
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 100%;
    height: 30px;
  }

  input {
    width: 70%;
    outline: none;
    height: 100%;
    border-radius: 5px;
    border: none;
    box-shadow: 0px 0px 3px 5px #00000024;
    padding-left: 10px;
  }

  button {
    width: 30%;
    height: 100%;
    border-bottom-right-radius: 5px;
    border-top-right-radius: 5px;
    border: none;
    background: #2196F3;
    color: #fff;
    position: relative;
    right: 5px;
    cursor: pointer;
    transition: all .5s ease;
    font-family: $fontRobotoCond;
    font-size: 1em;

    &:hover {
      background: #000;;
    }

  }

}
