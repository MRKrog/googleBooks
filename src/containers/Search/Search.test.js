import React, { Component } from 'react';
import { Search } from './Search';
import { shallow } from 'enzyme';

describe('Search', () => {
  let wrapper;
  let mockHandleSearch = jest.fn()
  let mockSearch = "";
  let event = { preventDefault: () => {}};

  beforeEach(() => {
    wrapper = shallow(
      <Search search={mockSearch}
              handleSearch={mockHandleSearch}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      search: "",
    });
  });

  it('should change state when handleInput method is invoked', () => {
    const input = wrapper.find('input');
    const mockEvent = { target: { name: 'search', value: 'Cooking' } }
    input.simulate('change', mockEvent);
    expect(wrapper.state()).toEqual({search: 'Cooking'});
  });

  it('should invoke handleSubmit when search button is clicked', () => {
    const spy = jest.spyOn(wrapper.instance(), "handleSubmit");
    wrapper.instance().forceUpdate();
    wrapper.find(".btnClick").simulate("click", event, spy);
    expect(spy).toHaveBeenCalled();
  });

  it('should invoke handleSearch with search parameters set in state', () => {
    wrapper.setState({
      search: 'Lock & Key'
    })
    const instance = wrapper.instance();
    instance.handleSubmit(event);
    expect(mockHandleSearch).toHaveBeenCalledWith('Lock & Key');

  });

  it('should invoke handleSearch and pass typed searched and then reset search state string', () => {
    wrapper.setState({
      search: 'Lock & Key'
    })
    const instance = wrapper.instance();
    instance.handleSubmit(event);
    expect(wrapper.state()).toEqual({search: ""})
  });
})
