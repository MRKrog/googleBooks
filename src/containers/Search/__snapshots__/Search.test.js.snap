// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
<div
  className="Search"
>
  <section
    className="searchTitle"
  >
    <h2>
      Google Books
    </h2>
  </section>
  <form
    className="searchForm"
  >
    <input
      name="search"
      onChange={[Function]}
      placeholder="Enter Your Search"
      type="search"
      value=""
    />
    <button
      className="btnClick"
      onClick={[Function]}
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
